compiler/rustc_ast_lowering/src/expr.rs

```diff
@@ -940,13 +956,17 @@ impl<'hir> LoweringContext<'_, 'hir> {
                 return self.pat_without_dbm(lhs.span, slice_pat);
             }
             // Tuple structs.
-            ExprKind::Call(callee, args) => {
-                if let ExprKind::Path(qself, path) = &callee.kind {
-                    let (pats, rest) =
-                        self.destructure_sequence(args, "tuple struct", eq_sign_span, assignments);
+            ExprKind::Call(callee, args) if self.is_tuple_struct(callee) => {
+                if let ExprKind::Path(None, path) = &callee.kind {
```

Comment:
Nit: the `ExprKind::Path(None, _)` check is already done in `is_tuple_struct`, if `is_tuple_struct` returned `Option<&Path>` instead of `bool`, then it would help to avoid duplication.

Reply:
Makes sense. Done.